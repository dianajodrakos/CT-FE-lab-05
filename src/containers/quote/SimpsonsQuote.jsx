/* eslint-disable max-len */
import React, { useState } from 'react';
import Quote from '../../components/quote/Quote';
import Load from '../../components/quote/Load';

const SimpsonsQuote = () => {
  const [loading, setLoading] = useState(false);
  const [quote, setQuote] = useState({});


  const handleClick = async () => {
    setLoading(true);
    getQuote()
      .then((quote) => setQuote(quote))
      .finally(() => setLoading(false));
  };

  if(loading) return <h1>Loading...</h1>;

  return (
    <>
      <Load onClick={handleClick} />
      <Quote quote={quote.quote} character={quote.character} image={quote.image} /> 
    </>
  );
};

export default SimpsonsQuote;
