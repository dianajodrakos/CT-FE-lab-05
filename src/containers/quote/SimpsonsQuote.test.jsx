/* eslint-disable max-len */
import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import SimpsonsQuote from '../../containers/quote/SimpsonsQuote';

const mockServer = setupServer(
  rest.get('https://thesimpsonsquoteapi.glitch.me/quotes', (req, res, ctx) => {
    return res(
      ctx.json([
        {
          quote: 'And this is the snack holder where I can put my beverage or, if you will, cupcake.',
          character: 'Homer Simpson',
          image: 'https://cdn.glitch.com/3c3ffadc-3406-4440-bb95-d40ec8fcde72%2FHomerSimpson.png?1497567511939',
        },
      ])
    );
  })
);

describe('SimpsonsQuote test', () => {
  beforeAll(() => mockServer.listen());
  afterAll(() => mockServer.close());

  it('should click a button on the DOM and reload the page with a random Simpsons character quote', async () => {
    render(<SimpsonsQuote />);

    const submitButton = screen.getByRole(
      'button',
      { name: 'get-quote' }
    );

    fireEvent.click(submitButton);

    return waitFor(() => {
      screen.getByText('Homer Simpson: And this is the snack holder where I can put my beverage or, if you will, cupcake.');
    });
  });
});
