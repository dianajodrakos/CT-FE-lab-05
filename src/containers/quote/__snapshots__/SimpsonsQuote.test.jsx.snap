// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpsonsQuote test should click a button on the DOM and reload the page with a random Simpsons character quote 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-label="get-quote"
      >
        Get a Quote
      </button>
      <h2>
        Loading...
      </h2>
    </div>
  </body>,
  "container": <div>
    <button
      aria-label="get-quote"
    >
      Get a Quote
    </button>
    <h2>
      Loading...
    </h2>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
